aws/resource_aws_rds_cluster.go

```diff
@@ -711,6 +757,126 @@ func resourceAwsRDSClusterCreate(d *schema.ResourceData, meta interface{}) error
 			return err
 		}
 
+	} else if v, ok := d.GetOk("restore_in_time"); ok {
+		pointInTime := v.([]interface{})[0].(map[string]interface{})
+
+		restoreToTimeAttr := pointInTime["restore_to_time"].(string)
+		useLatestRestorableTimeAttr := pointInTime["use_latest_restorable_time"].(bool)
+
+		if useLatestRestorableTimeAttr && restoreToTimeAttr != "" {
+			return fmt.Errorf(`provider.aws: aws_rds_cluster: %s: Either "restore_to_time" or "use_latest_restorable_time" must be set`, d.Get("database_name").(string))
+		}
+
+		createOpts := &rds.RestoreDBClusterToPointInTimeInput{
+			DBClusterIdentifier:       aws.String(identifier),
+			DeletionProtection:        aws.Bool(d.Get("deletion_protection").(bool)),
+			SourceDBClusterIdentifier: aws.String(pointInTime["source_cluster_identifier"].(string)),
+			Tags:                      tags,
+		}
+
```

Comment:
I think extracting it into a function only moves the code from this point to another without any additional benefit or reusability. The attribute `restore_to_point_in_time` get some safety guarantees from schema validations:

```
Type: schema.TypeList,
MaxItems: 1,
Elem: &schema.Resource{},
```

These three attributes together ensure that the attribute is decoded as a block and has at most 1 item. Further down the line `v, ok := d.GetOk("restore_to_point_in_time")` check ensures that the block is declared in code, and schema validation will prevent and empty value on it.

Reply:
fair point @Gufran !  instead of moving code that is only used in this part of the resource's create func, I would suggest then pulling the map values safely with the `v, ok := map[key].(string)` pattern (similar to what you have in L717 when extracting `restore_type`). this helps to ensure we don't hit unexpected panics and allows us to check if exactly 1 of the 2 possible attributes is set; we should be safe to assume that both `restore_to_time` and `use_latest_restorable_time` cannot be set at the same time as the `ConflictsWith` field is set on these attributes 👍 

e.g. 
```go
pointInTime := v.([]interface{})[0].(map[string]interface{})

if v, ok := pointInTime["restore_to_time"].(string); ok && v != "" {
  restoreToTime, err := time.Parse(time.RFC3339, restoreToTimeAttr)
  if err == nil {
    createOpts.RestoreToTime = aws.Time(restoreToTime)
  }
} 
if v, ok := pointInTime["use_latest_restorable_time"].(bool); ok && v {
  createOpts.UseLatestRestorableTime = aws.Bool(v)
}

if createOpts.RestoreToTime == nil && createOpts.UseLatestRestorableTime == nil {
  return fmt.Errorf(`provider.aws: aws_rds_cluster: %s: Either "restore_to_time" or "use_latest_restorable_time" must be set`, d.Get("database_name").(string))
}
```

Reply:
I should note though that the AWS RDS API is pretty good about notifying the user if one of these 2 fields is not set e.g. when neither param is sent 
```
Error: InvalidParameterValue: Please specify RestoreToTime or LatestRestorableTime
	status code: 400, 
```